sphinx/writers/latex.py

```diff
@@ -316,18 +315,49 @@ def is_footnote_ref(node):
 
 
 class Table(object):
-    def __init__(self):
-        # type: () -> None
+    def __init__(self, node):
+        # type: (nodes.table) -> None
+        self.header = []                        # type: List[unicode]
+        self.body = []                          # type: List[unicode]
+        self.classes = node.get('classes', [])  # type: List[unicode]
         self.col = 0
         self.colcount = 0
-        self.colspec = None             # type: unicode
-        self.colwidths = []             # type: List[int]
+        self.colspec = None                     # type: unicode
+        self.colwidths = []                     # type: List[int]
         self.rowcount = 0
-        self.had_head = False
         self.has_problematic = False
         self.has_verbatim = False
-        self.caption = None             # type: List[unicode]
-        self.longtable = False
+        self.caption = None                     # type: List[unicode]
+
+    def is_longtable(self):
+        # type: () -> bool
+        return self.rowcount > 30 or 'longtable' in self.classes
+
+    def get_table_type(self):
+        # type: () -> unicode
+        if self.is_longtable():
+            return 'longtable'
+        elif self.has_verbatim:
+            return 'tabular'
+        elif self.has_problematic and not self.colspec:
```

Comment:
the  ``table having :widths: option`` uses tabulary because of that but it should use tabular

in my PR  I had something like
```
        elif self.table.has_problematic or \
             (self.table.colwidths and 'colwidths-given' in self.table.classes):
```

other remark is that the templates produces extra blank lines in tex output. mentioning it here because I am off now for a while, will resume later ;-) congrats on great work ! ( `:align:`  option will cause evolutions)